File: AD-HOC/1087 - Queen/1087.cs
using System; 

class URI {

    static void Main(string[] args) {
        while (true)
        {
            string[] input = Console.ReadLine().Split(' ');
            int x1 = int.Parse(input[0]);
            int y1 = int.Parse(input[1]);
            int x2 = int.Parse(input[2]);
            int y2 = int.Parse(input[3]);

            if (x1 == 0 && y1 == 0 && x2 == 0 && y2 == 0)
                break;

            int dx = Math.Abs(x2 - x1);
            int dy = Math.Abs(y2 - y1);

            if (dx == 0 && dy == 0)
                Console.WriteLine(0);
            else if (dx == dy || dx == 0 || dy == 0)
                Console.WriteLine(1);
            else
                Console.WriteLine(2);
        }
    }
}

File: AD-HOC/1087 - Queen/1087.py
while True:
    input_data = input().split()
    x1 = int(input_data[0])
    y1 = int(input_data[1])
    x2 = int(input_data[2])
    y2 = int(input_data[3])

    if x1 == 0 and y1 == 0 and x2 == 0 and y2 == 0:
        break

    dx = abs(x2 - x1)
    dy = abs(y2 - y1)

    if dx == 0 and dy == 0:
        print(0)
    elif dx == dy or dx == 0 or dy == 0:
        print(1)
    else:
        print(2)